code/datums/datacore.dm

```diff
@@ -223,17 +273,9 @@ GLOBAL_DATUM_INIT(data_core, /datum/datacore, new)
 
 		var/static/record_id_num = 1001
 		var/id = num2hex(record_id_num++,6)
-		var/image = get_id_photo(H, show_directions)
-		var/datum/picture/pf = new
-		var/datum/picture/ps = new
-		pf.picture_name = "[H]"
-		ps.picture_name = "[H]"
-		pf.picture_desc = "This is [H]."
-		ps.picture_desc = "This is [H]."
-		pf.picture_image = icon(image, dir = SOUTH)
-		ps.picture_image = icon(image, dir = WEST)
-		var/obj/item/photo/photo_front = new(null, pf)
-		var/obj/item/photo/photo_side = new(null, ps)
+		// We need to compile the overlays now, otherwise we're basically copying an empty icon.
+		COMPILE_OVERLAYS(H)
+		var/mutable_appearance/character_appearance = new(H.appearance)
```

Comment:
you should use `build_appearance_list` for this,  so if anything new ever needs to be handed it can be done in one place
it will return a list mind, so you'll need to account for that

Reply:
After we chatted on Discord, we decided to leave this up to Lemon for a future refactor, as that's quite a bit more work for this PR.